src/tools/run-make-support/src/rustc.rs

```diff
@@ -85,6 +92,35 @@ impl Rustc {
         self
     }
 
+    /// Specify number of codegen units
+    pub fn codegen_units(&mut self, units: usize) -> &mut Self {
+        self.cmd.arg(format!("-Ccodegen-units={units}"));
+        self
+    }
+
+    /// Specify directory path used for incremental cache
+    pub fn incremental(&mut self, path: impl AsRef<Path>) -> &mut Self {
+        let path = path.as_ref().to_string_lossy();
+        self.cmd.arg(format!("-Cincremental={}", path));
+        self
+    }
+
+    /// Specify error format to use
+    pub fn error_format(&mut self, format: ErrorFormat) -> &mut Self {
```

Comment:
minor nit: for the API design, we experimented with it and ended up with a weakly-typed API with helper methods (see #122460), so this can just be

```rs
error_format(&mut self, format: &str)
```